### PR TITLE
ITM-195: Read in Metrics Eval scenario format

### PR DESCRIPTION
**Pull request by nextcen-dgemoets**
The following was accomplished for this PR:
- Updated API swagger to latest and re-generate model files
- Updated soartech and/or adept scenario for new format (adapted from [latest example](https://nextcentury.atlassian.net/wiki/download/attachments/3041951763/phase1-scenario.yaml?api=v2))
- Updated `ITMScenarioReader` to read new scenario file format and populate new model objects, including scenes
- Replaced probe reader with scene reader
- Deleted September-style scenarios and probes

To test:
- Connect a client to a running server and inspect the contents of the scenario and scenes after the call to `ITMSessionScenarioObjectHandler.read_scenario_from_yaml()`.  You can print them out, use a debugger, or whatever you wish. 
The server should reply to the client, but the client will crash because it hasn't been updated.
- Be sure to specify a `soartech` and `adept` client session (e.g., `--session soartech`), but not `eval`.  The adept scenario is the sample yaml file we've been maintaining, and the soartech scenario is a modified version of an pre-release scenario that soartech has provided to us.

### CODE REVIEW

**Review by kaitlyn-sharo — COMMENTED** [review]
Looks mostly good. The values seem to match, just a few small formatting things with the output that I'm not sure matter, though the unstructured text issues make it difficult to print everything nicely.

SOARTECH: 
The unstructured text has a lot of quotation marks and separations. Do we want that or do we want it to all be combined into a single string? For example, decision_environment unstructured text comes out exactly as "Isolated ""jungle ""island. ""EVAC is ""scheduled ""for 5 ""minutes.\n".

The same thing seems to be happening with the military_branch in demographics ("US Marine ""Corps"). I don't know if it's just the way I have it printing, or if it will matter, I just want to bring attention to it in case.


ADEPT:
It looks like some of the actions might have weird quotes going on. In the action list, I'm seeing 'action1', 'action2"', with an extra double quote.

File: swagger_server/itm/itm_scene.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    def __init__(self, action :Action, scene_index :int):
+        self.action_id = action.action_id
+        self.action_type = action.action_type
+        self.unstructured = action.unstructured
+        self.character_id = action.character_id
+        self.probe_id = action.probe_id
+        self.choice = action.choice
+        self.kdma_association = action.kdma_association
+        self.conditions = action.conditions
+        self.next_scene = scene_index+1 if action.next_scene == None else action.next_scene
```
[the remark this answers is not in the document]
Comment (nextcen-dgemoets) [reply swagger_server/itm/itm_scene.py]: Thanks, fixed!

File: swagger_server/itm/itm_scene.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    Class for managing a scene in the ITM system.
+    """
+
+    def __init__(self, scene :Scene):
+        """
+        Initialize an instance of ITMScene.
+        """
+        self.index = scene.index
+        self.state :State = scene.state
+        self.end_scene_allowed = scene.end_scene_allowed
```
[the remark this answers is not in the document]
Comment (nextcen-dgemoets) [reply swagger_server/itm/itm_scene.py]: Yep, can't help but add types more than I need to.  My Java background at work.  I added those types for consistency here.

**Comment by nextcen-dgemoets** [comment]
> The unstructured text has a lot of quotation marks and separations. Do we want that or do we want it to all be combined into a single string? For example, decision_environment unstructured text comes out exactly as "Isolated ""jungle ""island. ""EVAC is ""scheduled ""for 5 ""minutes.\n".

I think the weird state formatting is okay-- when I just print out, e.g., `military_branch`, I get "US Marine Corps".  I think it's some weird 80-character folding algorithm.

**Review by kaitlyn-sharo — APPROVED** [review]
Seems good to me!

**Comment by jaudick** [comment]
@nextcen-dgemoets 
Wow this is a big pr and looks really good. I love all of the new properties.

The server seems to be functioning ok. When I print out the new session on the server it looks correct. But the client minimal runner does not work. I am getting the same error for both adept and soartech. Am I doing something wrong?

## Server output for **adept** then **soartech**

```
(venv) jacob.audick_cn@K6L7J7HTVG itm-evaluation-server % python -m swagger_server
/Users/jacob.audick_cn/Documents/itm-evaluation-server/swagger_server/__main__.py:14: DeprecationWarning: 'app.json_encoder' is deprecated and will be removed in Flask 2.3. Customize 'app.json_provider_class' or 'app.json' instead.
  app.app.json_encoder = encoder.JSONEncoder
 * Serving Flask app '__main__'
 * Debug mode: off
WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:8080
 * Running on http://172.16.65.17:8080
Press CTRL+C to quit
127.0.0.1 - - [29/Jan/2024 11:32:50] "GET /ta2/startSession?adm_name=Testing123&session_type=adept&kdma_training=False&max_scenarios=1 HTTP/1.1" 200 -
--> Got alignment target from TA1.
127.0.0.1 - - [29/Jan/2024 11:32:50] "GET /ta2/scenario?session_id=be47d687-b1bb-4e21-b629-90c86affa943 HTTP/1.1" 200 -
--> Re-using session be47d687-b1bb-4e21-b629-90c86affa943 for ADM Testing123
127.0.0.1 - - [29/Jan/2024 11:33:33] "GET /ta2/startSession?adm_name=Testing123&session_type=soartech&kdma_training=False&max_scenarios=1 HTTP/1.1" 200 -
--> Got alignment target from TA1.
127.0.0.1 - - [29/Jan/2024 11:33:33] "GET /ta2/scenario?session_id=8978d8f2-a72f-4ff3-9541-da8eea307360 HTTP/1.1" 200 -
```




## Adept Client
### python3 itm_minimal_runner.py --adm_name Testing123 --session adept 1
```
(venv) jacob.audick_cn@K6L7J7HTVG itm-evaluation-client % python3 itm_minimal_runner.py --adm_name Testing123 --session adept 1
Traceback (most recent call last):
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/itm_minimal_runner.py", line 191, in <module>
    main()
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/itm_minimal_runner.py", line 162, in main
    scenario: Scenario = itm.start_scenario(session_id)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/swagger_client/api/itm_ta2_eval_api.py", line 467, in start_scenario
    (data) = self.start_scenario_with_http_info(session_id, **kwargs)  # noqa: E501
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/swagger_client/api/itm_ta2_eval_api.py", line 530, in start_scenario_with_http_info
    return self.api_client.call_api(
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/swagger_client/api_client.py", line 316, in call_api
    return self.__call_api(resource_path, method,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/swagger_client/api_client.py", line 160, in __call_api
    return_data = self.deserialize(response_data, response_type)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/swagger_client/api_client.py", line 232, in deserialize
    return self.__deserialize(data, response_type)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/swagger_client/api_client.py", line 271, in __deserialize
    return self.__deserialize_model(data, klass)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/swagger_client/api_client.py", line 618, in __deserialize_model
    kwargs[attr] = self.__deserialize(value, attr_type)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/swagger_client/api_client.py", line 271, in __deserialize
    return self.__deserialize_model(data, klass)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/swagger_client/api_client.py", line 618, in __deserialize_model
    kwargs[attr] = self.__deserialize(value, attr_type)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/swagger_client/api_client.py", line 271, in __deserialize
    return self.__deserialize_model(data, klass)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/swagger_client/api_client.py", line 620, in __deserialize_model
    instance = klass(**kwargs)
               ^^^^^^^^^^^^^^^
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/swagger_client/models/environment.py", line 81, in __init__
    self.unstructured = unstructured
    ^^^^^^^^^^^^^^^^^
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/swagger_client/models/environment.py", line 130, in unstructured
    raise ValueError("Invalid value for `unstructured`, must not be `None`")  # noqa: E501
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: Invalid value for `unstructured`, must not be `None`
(venv) jacob.audick_cn@K6L7J7HTVG itm-evaluation-client % 
```





## Soartech Client
### python3 itm_minimal_runner.py --adm_name Testing123 --session soartech 1
```
(venv) jacob.audick_cn@K6L7J7HTVG itm-evaluation-client % python3 itm_minimal_runner.py --adm_name Testing123 --session soartech 1
Traceback (most recent call last):
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/itm_minimal_runner.py", line 191, in <module>
    main()
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/itm_minimal_runner.py", line 162, in main
    scenario: Scenario = itm.start_scenario(session_id)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/swagger_client/api/itm_ta2_eval_api.py", line 467, in start_scenario
    (data) = self.start_scenario_with_http_info(session_id, **kwargs)  # noqa: E501
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/swagger_client/api/itm_ta2_eval_api.py", line 530, in start_scenario_with_http_info
    return self.api_client.call_api(
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/swagger_client/api_client.py", line 316, in call_api
    return self.__call_api(resource_path, method,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/swagger_client/api_client.py", line 160, in __call_api
    return_data = self.deserialize(response_data, response_type)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/swagger_client/api_client.py", line 232, in deserialize
    return self.__deserialize(data, response_type)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/swagger_client/api_client.py", line 271, in __deserialize
    return self.__deserialize_model(data, klass)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/swagger_client/api_client.py", line 618, in __deserialize_model
    kwargs[attr] = self.__deserialize(value, attr_type)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/swagger_client/api_client.py", line 271, in __deserialize
    return self.__deserialize_model(data, klass)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/swagger_client/api_client.py", line 618, in __deserialize_model
    kwargs[attr] = self.__deserialize(value, attr_type)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/swagger_client/api_client.py", line 271, in __deserialize
    return self.__deserialize_model(data, klass)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/swagger_client/api_client.py", line 620, in __deserialize_model
    instance = klass(**kwargs)
               ^^^^^^^^^^^^^^^
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/swagger_client/models/environment.py", line 81, in __init__
    self.unstructured = unstructured
    ^^^^^^^^^^^^^^^^^
  File "/Users/jacob.audick_cn/Documents/itm-evaluation-client/swagger_client/models/environment.py", line 130, in unstructured
    raise ValueError("Invalid value for `unstructured`, must not be `None`")  # noqa: E501
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: Invalid value for `unstructured`, must not be `None`
(venv) jacob.audick_cn@K6L7J7HTVG itm-evaluation-client % 

```

**Comment by nextcen-dgemoets** [comment]
Yes, in the testing instructions, I mentioned that the server should reply to the client, but the client will crash because it hasn't been updated.  This PR is just about ensuring that the scenario data is read and stored correctly.

**Review by jaudick — APPROVED** [review]
Oops I missed that line in the instructions. Looks good then!

**Pull request merged**